triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -129,19 +124,29 @@ object TriggerActor {
               Behaviors.same
           }
 
+      val darMap = config.dar.all.toMap
+      val compiler = Compiler(darMap)
+      val compiledPackages =
+        PureCompiledPackages(darMap, compiler.compilePackages(darMap.keys)).right.get
+
       val acsQuery =
         LedgerClient
           .singleHost(config.ledgerConfig.host, config.ledgerConfig.port, clientConfig)
           .flatMap { client =>
-            val runner = new Runner(client, appId, config.party, config.dar)
-            val (triggerExpr, triggerTy, triggerIds) = runner.getTrigger(config.triggerId)
-            val converter = Converter(runner.compiledPackages, triggerIds)
-            val filter = runner.getTriggerFilter(converter, triggerExpr, triggerTy)
+            val trigger = Runner.getTrigger(compiledPackages, config.triggerId)
+            val runner = new Runner(
```

Comment:
This deliberately doesn’t use `fromDar` in preparation for dynamic package uploads.